batch/batch/front_end/front_end.py

```diff
@@ -1038,21 +1040,31 @@ async def create_batch(request, userdata):
 
     @transaction(db)
     async def insert(tx):
-        billing_projects = await query_billing_projects(tx, user=user, billing_project=billing_project)
-
-        if len(billing_projects) != 1:
-            assert len(billing_projects) == 0
+        bp = await tx.execute_and_fetchone('''
+SELECT billing_projects.status, billing_projects.limit
+FROM billing_project_users
+INNER JOIN billing_projects
+  ON billing_projects.name = billing_project_users.billing_project
+WHERE billing_project = %s AND user = %s
+LOCK IN SHARE MODE''', (billing_project, user))
+
+        if bp is None:
             raise web.HTTPForbidden(reason=f'Unknown Hail Batch billing project {billing_project}.')
-        assert billing_projects[0]['status'] is not None
-        if billing_projects[0]['status'] in {'closed', 'deleted'}:
+        if bp['status'] in {'closed', 'deleted'}:
             raise web.HTTPForbidden(reason=f'Billing project {billing_project} is closed or deleted.')
 
-        bp = billing_projects[0]
+        cost = await tx.execute_and_fetchone('''
+SELECT COALESCE(SUM(`usage` * rate), 0) AS cost
```

Comment:
This logic is not quite the same as what I have in `query_billing_projects`. This code ignores that the billing for really old batches is in the `msec_mcpu` field.

Reply:
Does this mean _get_batch currently has a bug?

Reply:
No. Because `batches.msec_mcpu` is passed through to `batch_record_to_dict` which does the right thing:

```
    cost = format_version.cost(record['msec_mcpu'], record['cost'])
    d['cost'] = cost
```

Reply:
ah, never mind I see how that avoids it.

Reply:
Ugh. OK, I need to eliminate msec_mcpu. I fear reading all those others tables will just restore a bunch of deadlocks.

Reply:
Yeah, if we touch every other batch in the billing project, we effectively serialize batch creation and stall all jobs trying to complete.